runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/options/DataflowPipelineOptionsTest.java

```diff
@@ -90,4 +93,40 @@ public void testUtf8UserNameAndApplicationNameIsNormalized() {
     options.setAppName("fəˈnɛtık əsoʊsiˈeıʃn");
     assertEquals("f00n0t0k00so0si0e00n-0i00nt00n000n0l0-1208190706", options.getJobName());
   }
+
+  @Test
```

Comment:
Added some asserts.

There is another change that validates gcpTempLocation and set it for TempStoragePrefix in DataflowRunner. I want to do that in a separate PR.

gcpTempLocation is still optional in DataflowPipelineOptions.